tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc

```diff
@@ -351,39 +328,48 @@ LogicalResult ConvertTFSubOp::matchAndRewrite(Operation *op,
   return success();
 }
 
-LogicalResult ConvertTFMulOp::matchAndRewrite(Operation *op,
-                                              PatternRewriter &rewriter) const {
+LogicalResult ConvertTFMulOp::matchAndRewrite(Operation* op,
+                                              PatternRewriter& rewriter) const {
```

Comment:
Please avoid formatting changes in unrelated part of the code, this makes the diff unreadable.

Reply:
@joker-eph, @stellaraccident:

Ugh.   Looking back to the initial patch from Suraj which had similar formatting issues which were fixed for us and this patch, I think there is some disagreement between how our editors (emacs 25.2.2 on Ubuntu18.04 w/clang-6, I believe) are interpreting clang-format -style=Google and what the Google clang-format environment _actually_ is.   

This is the formatting that was chosen by the default clang-format picked up from tensorflow/.clang-format in our environment.  Interestingly, if I remove that file and put in my own clang-format -style=Google -dump-config, I get yet another interpretation. 

Obviously, this will be an irritating problem going forward if we don't get to the bottom of what's causing it.  Does this sort of issue ring any bells?

Reply:
I think I roughly figured out what's happening here.   Could I kindly ask for a pass on this, just this one time?  

TL;DR; the original clang-formatting does is wrong on legalize_tf.cc

The initial TOSA commit was formatted using -style=Google, which uses the clang-format parameter DerivePointerAlignment: true.  It should have used the default .clang-format in tensorflow/.clang-format, which sets that parameter to false.  DerivePointerAlignment is one of those mind-boggling parameters that chooses the setting for PointerAlignment based on the dominant usage already in the file.  For most of the files, it chose PointerAlignment: left, but for legalize_tf.cc, it chose PointerAlignment: right instead.    

The only thing the default TensorFlow clang-format file does is disable this per-file behavior and format PointerAlignment: left unconditionally.  

So we're left with a few choices:
1. Commit this file with some unrelated, but correct clang reformatting (what we have now, modulo the other todos)
2. Commit this file while emulating the incorrect clang reformatting with a manual override on clang-format (and possible validation failures later on?)
3. Insert and merge a separate commit before this one that corrects the issue, merge, rebase.

My preference is for (1) because it's simpler mechanically, although it does have the ugly reformatting + code changes.
We should definitely scrub the files that weren't touched in this commit to see if there are any other affected files.

Reply:
I'm fine with the first option. Dueling tools are not worth the terrible to diagnose in flight at this level. When you're ready, I can commit this and apply any formatting fixups as part of that.

Reply:
In general we use *git clang-format* and never format the entire file (so don't use editor full-file formatting): so it does not matter much that you have some local discrepancies with the clang-format version. The point is more about not exploding the size of the diff.